core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidator.java

```diff
@@ -129,10 +133,28 @@ private void throwBadRequest(StackView stack) {
                 stack.getName(), stack.getStatus()));
     }
 
-    public void validateStackStatusForStartHostGroup(StackView stack) {
-        if (!(stack.isAvailable() || stack.isAvailableWithStoppedInstances())) {
-            throw new BadRequestException(format("Data Hub '%s' has '%s' state. Node group start operation is not allowed for this state.",
-                    stack.getName(), stack.getStatus()));
+    public void validateStackStatusForStartHostGroup(StackDto stack, InstanceGroupAdjustmentV4Request instanceGroupAdjustmentV4Request) {
+        CmTemplateProcessor processor = cmTemplateProcessorFactory.get(stack.getBlueprint().getBlueprintText());
+        Map<String, String> roleDependencyMap = roleDependencyService.getRoleDependency();
+        List<String> unhealthyHostsGroupNames = processor.getUnhealthyDependentHostGroups(stack, roleDependencyMap);
+        if (stack.getStack().isModificationInProgress()) {
+            throw new BadRequestException(format("Data Hub '%s' has '%s' state. Upscaling is not allowed.",
+                    stack.getStack().getName(), stack.getStack().getStatus()));
+        }
+
+        if (!unhealthyHostsGroupNames.isEmpty()) {
+            if (unhealthyHostsGroupNames.size() > 1) {
+                throw new BadRequestException(format("Upscaling is Not Allowed for HostGroup: '%s' as Data hub '%s' has " +
+                        unhealthyHostsGroupNames.stream().filter(i -> !i.equals("noDependentHostGroupFound")).collect(Collectors.toList()) +
+                        " Node Unhealthy.", instanceGroupAdjustmentV4Request.getInstanceGroup(), stack.getStack().getName()));
```

Comment:
What if there are multiple entries with just `noDependentHostGroup`? Correct me if I'm wrong here, but the flow should proceed if we cannot infer the health of the dependent hosts right?

Reply:
If there are multiple entries with noDependentHostGroup, then it that case as well it will consider it only once as I have used a Set for storing them.
Yes, the flow will proceed if we cannot infer the health of the dependent hosts.